File: game.of.life/game.of.life.domain/Cycle.cs
namespace game.of.life.domain
{
    using System.Linq;

    public class Cycle
    {

        public RectangularInfinite2DGrid Run(RectangularInfinite2DGrid currentGrid)
        {
            var newGrid = InitializeNewGrid(currentGrid);

            ComputeAndCompleteMutation(newGrid);

            PrepareForNextCycle(newGrid);

            return newGrid;
        }

        private static RectangularInfinite2DGrid InitializeNewGrid(IGrid currentGrid)
        {
            var newGrid = new RectangularInfinite2DGrid(currentGrid.Cells.Select(c => new Cell(c.X, c.Y, c.State)));
            newGrid.Discover();
            return newGrid;
        }

        private static void ComputeAndCompleteMutation(IGrid newGrid)
        {
            newGrid.Cells.ToList().ForEach(cell => cell.ComputeNextMutation(newGrid.GetNeighbours(cell).Count(c => c.IsAlive())));
            newGrid.Cells.ToList().ForEach(cell => cell.CompleteMutation());
        }

        private static void PrepareForNextCycle(IGrid newGrid)
        {
            newGrid.Discover();
            newGrid.Clean();
        }
    }
}

File: game.of.life/ts.old/game.of.life.domain.ts/Cycle.ts
class Cycle {

    run(currentGrid : RectangularInfinite2DGrid): RectangularInfinite2DGrid {
        var newGrid = this.initializeNewGrid(currentGrid);
        this.computeAndCompleteMutation(newGrid);
        this.prepareForNextCycle(newGrid);
        return newGrid;
    }

    private initializeNewGrid(currentGrid : RectangularInfinite2DGrid): RectangularInfinite2DGrid {
        var newGrid = new RectangularInfinite2DGrid(currentGrid.Cells.ToList<Cell>().Select(c => new Cell(c.x, c.y, c.state)).ToArray());
        newGrid.discover();
        return newGrid;
    }

    private computeAndCompleteMutation(newGrid : RectangularInfinite2DGrid): void {
        newGrid.Cells.ToList<Cell>().ForEach(cell => cell.computeNextMutation(newGrid.getNeighbours(cell).ToList<Cell>().Count(c => c.isAlive())));
        newGrid.Cells.ToList<Cell>().ForEach(cell => cell.completeMutation());
    }

    private prepareForNextCycle(newGrid : RectangularInfinite2DGrid): void {
        newGrid.discover();
        newGrid.clean();
    }
}